### PR TITLE
DOC: Docstring Redesign to fix the problem of unexpected keyword arg (issue10888)

### PR DESCRIPTION
**Pull request by terrytangyuan**
Fixed issue #10888

### CODE REVIEW

**Comment by terrytangyuan** [comment]
Should I add bug fix note in whatsnew?

**Comment by terrytangyuan** [comment]
@jreback @jorisvandenbossche  On green now. Could you take a look? This change won't break anything else.

**Comment by terrytangyuan** [comment]
@jreback  Is this one ok to merge?

**Comment by jreback** [comment]
needs tests

**Comment by terrytangyuan** [comment]
So this is simply a DOC problem where users use the wrong parameters specified in Documentation but they don't belong to those methods. I've fixed the documentation so users won't get confused again. I don't think unit tests are needed for this PR.

File: pandas/core/frame.py [hunk 1]
```diff
@@ -1494,7 +1494,7 @@ def to_html(self, buf=None, columns=None, col_space=None, colSpace=None,
         if buf is None:
             return formatter.buf.getvalue()
 
-    @Appender(fmt.docstring_to_string, indents=1)
+    @Appender(fmt.common_docstring + fmt.return_docstring, indents=1)
```
Comment (jreback) [inline pandas/core/frame.py]: can you make these all be separated by ``\n`` so this is consistent.

**Comment by jreback** [comment]
can you post before-after of the doc-strings

File: pandas/core/format.py [hunk 2]
```diff
@@ -52,20 +52,27 @@
...
-        formatting is now rendered to unicode by default.
+        formatting is now rendered to unicode by default."""
+
+return_docstring = """
 
     Returns
     -------
     formatted : string (or unicode, depending on data and options)"""
 
+docstring_to_string = common_docstring + "\n" + justify_docstring + "\n" + force_unicode_docstring + return_docstring
```
Comment (jorisvandenbossche) [inline pandas/core/format.py]: no `\n` is needed between commong_docstring and justify_docstring

Comment (jorisvandenbossche) [inline pandas/core/format.py]: btw, where is this variable used?

File: pandas/core/frame.py [hunk 3]
```diff
@@ -1416,7 +1416,7 @@ def to_stata(
                              write_index=write_index)
         writer.write_file()
 
-    @Appender(fmt.docstring_to_string, indents=1)
+    @Appender(fmt.common_docstring + "\n" + fmt.justify_docstring + fmt.return_docstring, indents=1)
```
Comment (jorisvandenbossche) [inline pandas/core/frame.py]: the `n` is not needed

File: pandas/core/format.py [hunk 4]
```diff
@@ -25,7 +25,7 @@
 import itertools
 import csv
 
-docstring_to_string = """
+common_docstring = """
      Parameters
      ----------
      frame : DataFrame
```
Comment (jorisvandenbossche) [inline pandas/core/format.py]: This is one space of (this was already, but can you fix it?

**Comment by jorisvandenbossche** [comment]
Apart from the few comments, looks good!

**Comment by terrytangyuan** [comment]
@jreback @jorisvandenbossche Thanks! They look better now!

File: pandas/core/format.py [hunk 5]
```diff
@@ -24,11 +24,11 @@
...
 
-docstring_to_string = """
-     Parameters
-     ----------
-     frame : DataFrame
-         object to render
+common_docstring = """
+    Parameters
+    ----------
+    frame : DataFrame
```
Comment (jreback) [inline pandas/core/format.py]: frame should come out, we don't normally document the type of the self paramater (and thus its confusing).

**Comment by terrytangyuan** [comment]
@jreback Thanks. Anything else I should change before I add a new burden to Travis?

**Comment by jreback** [comment]
thanks!

**Pull request merged**